vaadin-upload.html

```diff
@@ -151,7 +151,7 @@
       <div id="buttonsPrimary">
         <div id="addFiles" on-tap="_onAddFilesClick">
           <slot name="add-button">
-            <paper-button id="addButton" disabled$="[[maxFilesReached]]">
```

Comment:
lets introduce a new boolean property called allowOverwrite and disable the button only when maxFilesReached and !allowOverwrite, you could use the same logic everywhere so we dont break the exciting functionality.